src/packages/cli/src/args.ts

```diff
@@ -177,5 +180,22 @@ export default function (version: string, isDocker: boolean) {
     .wrap(wrapWidth)
     .version(version);
 
-  return (args as unknown) as Yargs;
+  const parsedArgs = args.argv;
+  const finalArgs = {
+    flavor: parsedArgs._.length > 0 ? parsedArgs._[0] : DefaultFlavor
+  } as Argv;
+  for (let key in parsedArgs) {
+    const parts = key.split(".");
+    // only copy namespaced keys
+    if (parts.length > 1) {
+      const last = parts.pop();
+      const obj = parts.reduce((acc: any, part: string) => {
+        // build the obj path (e..g, `finalArgs: {wallet: {}}`)
+        return (acc[part] = acc[part] || {});
+      }, finalArgs);
+      obj[last] = parsedArgs[key];
```

Comment:
FWIW, this is a bit non-intuitive

I would have written something like

```suggestion
      const [ optionCategory, optionName ] = parts;
      if (!finalArgs[optionsCategory]) {
        finalArgs[optionsCategory] = {};
      }
      finalArgs[optionsCategory][optionName] = parsedArgs[key];
```

Up to you how you want to portray this logic

Reply:
Mine was just supporting a possible future where there are more than one "." in a key name. A contrived example: `"server.ipfs.port"` would create `server:{ ipfs: {port: 1234} }`.

So, the logic between the two is a little different; yours only takes into account what is possible with our current options... which is probably okay, since tons of other internals would have to be rewritten to support that future.

So...I'll switch to yours. :-) Commit incoming!

Reply:
updated: https://github.com/trufflesuite/ganache-core/pull/721/files#diff-2b2281ab4da879ef24d594d2dc8cbee929624f9b3e1438f88d20694aa0351490R184-R200